### PR TITLE
Connect Report Definition Details Backend

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*

*Description of changes:*
Connect Report Definition Details to back-end to provide actual details of each Report definition.

`Note`: Depends on #51 

`TODO`: Update data model/schema to include more fields specifying differences between email and Kibana recipients 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/components/main/report_definition_details/report_definition_details.tsx [hunk 1]
```diff
@@ -30,62 +30,57 @@ import {
...
+    let reportDefinitionDetails = {
+      name: data['report_name'],
+      description: data['description'],
+      created: data['time_created'],
+      last_updated: data['last_updated'],
+      source: data['report_source'],
+      time_period:
+        data['trigger']['trigger_params']['schedule']['interval']['period'] +
+        data['trigger']['trigger_params']['schedule']['interval']['days'],
+      file_format: data['report_params']['report_format'],
```
Comment (mihirsoni) [inline public/components/main/report_definition_details/report_definition_details.tsx:51]: Be consist on the naming, I think mostly it is `camelCase` Use everywhere the same case.

File: public/components/main/report_definition_details/report_definition_details.tsx [hunk 2]
```diff
@@ -30,62 +30,57 @@ import {
...
-}
 
-export function ReportDefinitionDetails(
-  props: ReportDefinitionDetailsRouteProps
-) {
-  const reportId = props.reportDefinitionId;
+  useEffect(() => {
+    const { httpClient } = props;
+    httpClient
+      .get('../api/reporting/reportDefinitions/' + reportDefinitionId)
```
Comment (mihirsoni) [inline public/components/main/report_definition_details/report_definition_details.tsx:72]: Use template string for concatenation. 

```js
.get(`../api/reporting/reportDefinitions/${repordId}`);
```

File: public/components/main/report_definition_details/report_definition_details.tsx [hunk 3]
```diff
@@ -101,7 +96,7 @@ export function ReportDefinitionDetails(
             <EuiFlexItem>
               <EuiPageHeaderSection>
                 <EuiTitle>
-                  <h2>{reportDefinitionDetailsMockMetadata.name}</h2>
+                  <h2>{reportDefinitionDetails['name']}</h2>
```
Comment (mihirsoni) [inline public/components/main/report_definition_details/report_definition_details.tsx:99]: Any reason why using bracket over `dot(.)` notiation?

Comment (davidcui-amzn) [reply public/components/main/report_definition_details/report_definition_details.tsx:102]: I always get error thrown when I try with dot notation- am I not constructing the object correctly?

File: public/components/main/report_definition_details/report_definition_details.tsx [hunk 4]
```diff
@@ -30,62 +30,60 @@ import {
...
+    const { httpClient } = props;
+    httpClient
+      .get(`../api/reporting/reportDefinitions/${reportDefinitionId}`)
+      .then((response) => {
+        handleReportDefinitionDetails(
+          getReportDefinitionDetailsMetadata(response)
+        );
+      })
+      .catch((error) => {
+        console.log('error when getting report definition details:', error);
```
Comment (joshuali925) [inline public/components/main/report_definition_details/report_definition_details.tsx:80]: Would it be better to use console.error here?

File: public/components/main/report_definition_details/report_definition_details.tsx [hunk 5]
```diff
@@ -110,11 +108,11 @@ export function ReportDefinitionDetails(
...
-              <EuiFlexItem></EuiFlexItem>
-              <EuiFlexItem></EuiFlexItem>
-              <EuiFlexItem></EuiFlexItem>
-              <EuiFlexItem></EuiFlexItem>
-              <EuiFlexItem></EuiFlexItem>
+              <EuiFlexItem />
+              <EuiFlexItem />
+              <EuiFlexItem />
+              <EuiFlexItem />
+              <EuiFlexItem />
```
Comment (joshuali925) [inline public/components/main/report_definition_details/report_definition_details.tsx:115]: Could you change line 142, 143 as well just to be consistent?

Comment (davidcui-amzn) [reply public/components/main/report_definition_details/report_definition_details.tsx:115]: Good catch, missed that

File: public/components/main/report_definition_details/report_definition_details.tsx (hunk 4 above)
Comment (davidcui-amzn) [reply public/components/main/report_definition_details/report_definition_details.tsx:80]: Sure, will change

**Pull request merged**